### PR TITLE
nixos/blockbook-frontend: init

### PR DESCRIPTION
**Pull request by 1000101**
###### Motivation for this change
Initial definition of blockbook-frontend service - blockchain explorer, its test and a small change to the package itself so we can run this as a service with multiple instances and different blockchains.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/networking/blockbook-frontend.nix [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+  };
+in
+{
+  # interface
+
+  options = {
+    services.blockbook-frontend = mkOption {
+      type = types.attrsOf (types.submodule blockbookOpts);
+      default = {};
+      description = "Sepcification of one or more blockbook-frontend instances.";
```
Comment (prusnak) [inline nixos/modules/services/networking/blockbook-frontend.nix:195]:
```suggestion
      description = "Specification of one or more blockbook-frontend instances.";
```

File: nixos/modules/services/networking/blockbook-frontend.nix [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+      name = cfg.user;
+      group = cfg.group;
+      home = cfg.dataDir;
+      isSystemUser = true;
+    })) eachBlockbook;
+
+    users.groups = mapAttrs' (instanceName: cfg: (
+      nameValuePair "${cfg.group}" { })) eachBlockbook;
+  };
+}
```
Comment (prusnak) [inline nixos/modules/services/networking/blockbook-frontend.nix:272]: Please add newline here

**Review by prusnak — CHANGES_REQUESTED** [review]

File: nixos/modules/services/networking/blockbook-frontend.nix (hunk 1 above)
Comment (1000101) [reply nixos/modules/services/networking/blockbook-frontend.nix]: Thanks!

**Comment by 1000101** [comment]
@ofborg test blockbook-frontend

**Comment by 1000101** [comment]
@ofborg test blockbook-frontend

**Comment by 1000101** [comment]
@ofborg test blockbook-frontend

**Comment by 1000101** [comment]
@ofborg test blockbook-frontend

**Comment by 1000101** [comment]
@ofborg test blockbook-frontend

**Comment by 1000101** [comment]
@ofborg test blockbook-frontend

**Comment by 1000101** [comment]
@ofborg test blockbook-frontend

**Comment by 1000101** [comment]
@ofborg test blockbook-frontend

**Comment by 1000101** [comment]
@ofborg test blockbook-frontend

**Comment by mmahut** [comment]
@ofborg test blockbook-frontend

**Comment by mmahut** [comment]
Thank you, good work.

**Pull request merged**